PWGHF/Tasks/HFMCValidation.cxx

```diff
@@ -173,15 +173,15 @@ struct ValidationRecLevel {
       }
       if (std::abs(candidate.flagMCMatchRec()) == 1 << hf_cand_prong2::DecayType::D0ToPiK) {
         if (candidate.index0_as<aod::BigTracksMC>().has_mcParticle()) {
-          indexParticle = RecoDecay::getMother(particlesMC, candidate.index0_as<aod::BigTracksMC>().mcParticle_as<aod::McParticles_000>(), pdg::Code::kD0, true);
+          indexParticle = RecoDecay::getMother(particlesMC, candidate.index0_as<aod::BigTracksMC>().mcParticle(), pdg::Code::kD0, true);
         }
         auto mother = particlesMC.iteratorAt(indexParticle);
         registry.fill(HIST("histPt"), candidate.pt() - mother.pt());
         registry.fill(HIST("histPx"), candidate.px() - mother.px());
         registry.fill(HIST("histPy"), candidate.py() - mother.py());
         registry.fill(HIST("histPz"), candidate.pz() - mother.pz());
         //Compare Secondary vertex and decay length with MC
-        auto daughter0 = mother.daughter0_as<aod::McParticles_000>();
+        auto daughter0 = particlesMC.iteratorAt(mother.daughtersIds().front());
```

Comment:
Does `mother.daughters().front()` work here?

Reply:
This does not work, but I will update it with the method to retrieve the first daughter implemented in https://github.com/AliceO2Group/AliceO2/pull/8113

Reply:
The previous comment was not correct: as discussed in person I needed to use `begin()` instead of `front()` so it works now